Makefile

```diff
@@ -5,6 +5,19 @@ DOCKER_IMAGE_REPO ?= quay.io/thanos/thanos
 DOCKER_IMAGE_TAG  ?= $(subst /,-,$(shell git rev-parse --abbrev-ref HEAD))-$(shell date +%Y-%m-%d)-$(shell git rev-parse --short HEAD)
 DOCKER_CI_TAG     ?= test
 
+SHA=''
+arch = $(shell uname -m)
+# Run `docker manifest inspect quay.io/prometheus/busybox:latest` to get SHA
+ifeq ($(arch), x86_64)
+    # amd64
+    SHA="0c38f63cbe19e40123668a48c36466ef72b195e723cbfcbe01e9657a5f14cec6"
+else ifeq ($(arch), armv8)
+	# arm64
+    SHA="3ab46d9f23bc36c511f3206ebdd40bfb5a63663acc0f3fc02ca24f5dbfb762a2"
```

Comment:
I think these may be a little cumbersome to maintain and error prone

Reply:
Consider something more automatic like

```make
SHA=$(shell docker manifest inspect $(BASE_IMAGE) | jq '.manifests[] | select(.platform.architecture == "$(arch)'") | if .platform | has("variant") then select(.platform.variant == "'$(variant)") else . end | .digest' -r)
```

Reply:
It is hard to do this, please look at https://quay.io/repository/prometheus/busybox?tag=latest&tab=tags.
Busybox has no tag expect latest or SHA.
We want to pin the busybox version, we can not use latest, the SHA is the only choice.
Since SHA is diffienert for each arch, hardcode here maybe look ugly but fit this case.

Reply:
We can comment here about how to get the SHA.

Reply:
The code I shared with you automatically determines the correct sha for the selected arch. The problem is that simply hardcoding a sha value will definitely get out of date and be difficult to maintain. The wrong sha *will* be chosen at one point for an arch even if we write the steps down. Generating it automatically is the right thing to do IMO

Reply:
I know your code works, but the `BASE_IMAGE` does not exist indeed.
In above [comment](https://github.com/thanos-io/thanos/pull/1923#issuecomment-568635587), i use busybox:latest as `BASE_IMAGE`.
As i said, busybox has no tags expect `latest` or SHA, and SHA is diffienert for each arch. When busybox update, your code would update all SHA which does not meet our needs(We do not always want to update SHA when busybox update, do we?).

Reply:
I think generating it automatically is helpful when we confirm that we need to update busybox version. To use a fixed busybox version, hardcode is necessary.